ghga_connector/core/file_operations.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2021 - 2022 Universität Tübingen, DKFZ and EMBL
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Contains Calls of the Presigned URLs in order to Up- and Download Files
+"""
+
+import pycurl
+
+from .exceptions import BadResponseCodeError, RequestFailedError
+
+
+def download_file(download_url, output_file):
+    """Download File"""
+
+    with open(output_file, "wb") as file:
```

Comment:
What if the call site does not sanity check that the file is actually a file and not a folder, does this produce the correct/a meaningful exception?